inference-engine/tests/functional/plugin/shared/include/behavior/core_threading_tests.hpp

```diff
@@ -215,6 +223,204 @@ class CoreThreadingTestsWithIterations : public ::testing::TestWithParam<CoreThr
 
     unsigned int numIterations;
     unsigned int numThreads;
+    bool isGnaTest = false;
+
+    // TODO: replace with subgraph builders after fixing *-31414
+    const std::vector<FuncTestUtils::TestModel::TestModel> defaultModels = {
+        FuncTestUtils::TestModel::convReluNormPoolFcModelFP32,
+        FuncTestUtils::TestModel::convReluNormPoolFcModelFP16
+    };
+
+    std::vector<InferenceEngine::CNNNetwork> networks;
+    void SetupNetworks(InferenceEngine::Core& ie) {
+        if (!isGnaTest) {
+            for (auto& model : defaultModels) {
+                networks.emplace_back(ie.ReadNetwork(model.model_xml_str, model.weights_blob));
+            }
+            networks.emplace_back(InferenceEngine::CNNNetwork(ngraph::builder::subgraph::make2InputSubtract()));
+            networks.emplace_back(InferenceEngine::CNNNetwork(ngraph::builder::subgraph::makeMultiSingleConv()));
+            networks.emplace_back(InferenceEngine::CNNNetwork(ngraph::builder::subgraph::makeSingleConv()));
+            networks.emplace_back(InferenceEngine::CNNNetwork(ngraph::builder::subgraph::makeSplitConvConcat()));
+            networks.emplace_back(InferenceEngine::CNNNetwork(ngraph::builder::subgraph::makeSplitMultiConvConcat()));
+        } else {
+            for (unsigned i = 0; i < numThreads; i++) {
+                networks.emplace_back(InferenceEngine::CNNNetwork(ngraph::builder::subgraph::makeConvPoolRelu()));
+            }
+        }
+    }
+
+    std::string GetModelXmlStr(unsigned index) {
+        if (isGnaTest) {
+            return GetSimpleGnaCompatModel();
+        }
+        return defaultModels[index % defaultModels.size()].model_xml_str;
+    }
+    InferenceEngine::Blob::Ptr GetModelWeightsBlob(unsigned index) {
+        if (isGnaTest) {
+            return CommonTestUtils::getWeightsBlob(1000000);
+        }
+        return defaultModels[index % defaultModels.size()].weights_blob;
+    }
+
+    std::string GetSimpleGnaCompatModel() {
+        return (R"V0G0N(
+<net batch="1" name="DNN_GNA_S" version="2">
```

Comment:
What is the difference of this model with the other one? Could you use NGraph function instead?

Reply:
The other one (models), is not compatible with GNA Plugin. 
And the model is not much relevant in this test as it is focused on MT not model itself.
Probably we would need it is PR3331 is merged first.

Reply:
i asked the same question. Please, use ngraph function. If it is necessary, merge your PR after 3331